### DISCUSSION TITLE
OBS SRT Streaming not working

### DISCUSSION THREAD

**Discussion by PanterSoft** [OvenMediaLabs/OvenMediaEngine#1194] [Help] [answered]
Hello,

I am trying to stream from OBS to OME over SRT. When I stream from my smartphone via SRT everything works fine but when using OBS This is not working. I watched the documentation from OBS and OME regarding SRT streaming and found out that some links in the documentation are no longer valid also I think the SRT streaming URL Format of OBS changed.

Does someone know the new valid streaming URL format for OME ?

**Comment by bchah** [comment]
It’s often related to the way the URL is escaped but it’s really hard to say without an example of what you are trying. Can you provide some more evidence, e.g. the SRT URL string you are using in OBS and the OME logs showing how it is receiving the request? This will give some better clues.

**Reply by PanterSoft** [reply]
Thanks for the response.

My SRT URL for OBS looks like this: 

- srt://MY_DOMAIN:MY_PORT?streamid=srt%3A%2F%2FMY_DOMAIN%3A9999%2Fapp%2FMY_STREAM_ID

Inside Larix Broadcaster the rolling SRT URL works fine but not in OBS

- srt://MY_DOMAIN:MY_PORT?streamid=srt://MY_DOMAIN:MY_PORT/app/MY_STREAM_ID[?query=value]

I am generating my Stream IDs based on a random hash generator. the stream IDs are looking like: "Iiprog=="

Here is my OME Log of the day I encountered the problems.

[ovenmediaengine.log](https://github.com/AirenSoft/OvenMediaEngine/files/11301847/ovenmediaengine.log)

**Reply by bchah** [reply]
Try unescaping the forward slashes. Here's a working example URL from my app:

srt://demo.net:9999?streamid=srt%3A//demo.net%3A9999/live/733a110e56525%3Fp%3DeyJ1cmxfZXhwaXJlIjo3MjQ1NTk0MTMwMDAwfQ%26s%3DdvZBffe8eCBV4X4kEDUlpoc_AXY

**Reply by PanterSoft** [reply]
Thanks for The help,

I will try on Monday when I am at home and can test it on my setup.

**Reply by PanterSoft** [reply]
Hi,

I tried it with no luck.

I made some further Investiagations to check if it is a problem with OBS or whatever. I can stream now via FFMPEG directly to OME with no problems and the encoding of the url semse to make no difference here (encoded and unencoded) but why is it not working with OBS

any further thoughts ?

P.S I looked at the log files again and I don't even get error messages or an information that OBS is trying to connect. but when using FFMpeg I get an response in the log file.

**Reply by getroot** [reply]
Could you try updating OBS to the latest version? Support for SRT in OBS is relatively recent.

**Reply by PanterSoft** [reply]
> Could you try updating OBS to the latest version? Support for SRT in OBS is relatively recent.
I got the newest version which is 29.0.2

**Reply by getroot** [reply]
My OBS version is also 29.0.2.
And I just tested the following and confirmed that it works normally.

![image](https://user-images.githubusercontent.com/17779085/234165948-eb143092-b39c-4256-b98a-0ce6f450f90b.png)

srt://192.168.0.160:19999?streamid=srt%3A%2F%2Fome-dev.airensoft.com%3A19999%2Fapp%2Fstream

**Reply by PanterSoft** [reply]
Okay Then I have to find another problem on my Side.

you are using Port 19999 but default port would be 9999 is that correct ?

**Comment by PanterSoft** [comment, marked as answer]
Here to share if someone else encounters the same Problem.

I found a problem but not the root or a potential fix.

If I use a domain in "srt://MY_DOMAIN:9999?streamid=..." its not working but however if I use the public IP address of the server like that "set://PUBLIC_IP:9999?streamid=..." it is working.

What I am confused about is that when I stream from an Smartphone with the domain version everything works as usual.

Furthermore it only works on macOS and not on Ubuntu 20.04

Based on this I assume this is a problem in OBS
